### ISSUE TITLE
Git Credentials Helper Support

### ISSUE THREAD

**Issue by jonwinton** [devopsmakers/xterrafile#2]
I'm using [Atlantis](https://github.com/runatlantis/atlantis) with Terraform and am able to use [git's credentials helper](https://git-scm.com/docs/gitcredentials) to authenticate when checking out modules from a private repo. This works when the remote module is referenced in the `source` field.

This doesn't work for [Xterrafile's checkout](https://github.com/devopsmakers/xterrafile/blob/9b663fe8b3683f8839cde4982432febbe4b60bf3/cmd/install.go#L165), but I'm wondering what the optimal solution might be. I'm happy to try and get a PR together but didn't know if there was something already considered for this.

**Comment by pysysops** [comment]
Hey @jonwinton thanks for raising the issue. 

I'll take a look at this very soon 👍

**Comment by ritsok** [comment]
@pysysops +1 for this!

**Comment by jeff-knurek** [comment]
I think I've run into a similar problem with  xterrafile's checkout.
Found this long running thread about `git.PlainClone()` authentication. 
https://github.com/src-d/go-git/issues/637#issuecomment-404851019

I don't know how this example solution will work with git's credentials helper, but might be a good place to start....

**Comment by jonwinton** [comment]
@pysysops sorry to bother, but if there's any way to help with this I'd be happy to pitch in.

**Comment by pysysops** [comment]
I guess the most pragmatic solution might be to fallback to using the `git` cli.

It's nice to have a self contained binary but in reality pretty much every OS comes with `git` out of the box. 🤔

**Comment by jonwinton** [comment]
Yeah, Terraform itself has a dependency on `git` since modules can be sourced from repos. I don't think it's an unsafe assumption to make. 

How are you thinking the fallback should be handled? If the checkout with the self contained `git` package fails then try to execute using the system's `git`?

**Comment by pysysops** [comment]
Hey @jonwinton @ritsok - Could you try out the latest version (v1.0.0) with Git credentials? I'v updated `xterrafile` to use Hashicorp's `go-getter` in a similar way to `terraform`.

This means that `xterrafile` is no longer using `go-git` and has a dependency on `git` being installed (which it is on most systems). It opens up `xterrafile` to all the module sources available to `terraform` (https://www.terraform.io/docs/modules/sources.html) accepting both the `terraform` source urls and the original `xterrafile` configs.

Any feedback would be appreciated :)

**Comment by jonwinton** [comment]
@pysysops Thank you so much, I'll test this out today! Can't tell you how excited I am for this, it's going to improve our workflow so much.
